pkg/repository/provider/restic.go

```diff
@@ -46,10 +47,19 @@ func (r *resticRepositoryProvider) ConnectToRepo(ctx context.Context, param Repo
 }
 
 func (r *resticRepositoryProvider) PrepareRepo(ctx context.Context, param RepoParam) error {
-	if err := r.InitRepo(ctx, param); err != nil {
+	if err := r.ConnectToRepo(ctx, param); err != nil {
+		// If the repository has not yet been initialized, the error message will always include
+		// the following string. This is the only scenario where we should try to initialize it.
+		// Other errors (e.g. "already locked") should be returned as-is since the repository
+		// does already exist, but it can't be connected to.
+		if strings.Contains(err.Error(), "Is there a repository at the following location?") {
+			return r.InitRepo(ctx, param)
```

Comment:
You don't need to call `ConnectToRepo` in this case?

Reply:
Confirmed that when the `InitRepo` is called the Connect is no necessaary.

Reply:
Not any more. The syntax of the caller's requirement (BackupRepository Controller's `ensureRepo` function) is: connect the repo first, if it is not initialized, initialize it, or simply speaking, make sure the repo is usable, and the newly added method PrepareRepo is for this purpose.